### PR TITLE
IPC Json

### LINKED ISSUE #223: Problem: The `mina-indexer` client server protocol serialization in binary
In order to unlock the full potential for the client server protocol, it should be changed to be texted based. Replace `BCS` serialization with JSON as the network payload. This enables us to use unix cli tooling like `nc` and `jq`

### PR DESCRIPTION
**Pull request by jenr24G**
Fix #223: Changes the ser and deser used for the client/server's IPC channel to use `serde_json` instead of `bcs`

### CODE REVIEW

**Comment by trevorbernard** [comment]
Add some tests to exercise the new found goodness

File: test [hunk 1]
```diff
@@ -36,11 +79,27 @@ idxr server config 2>&1 |
...
 
-# Indexer server config parses example yaml config file
+# Indexer server config passes cli with minimal args
 idxr server cli \
     --initial-ledger tests/ledgers/mainnet-genesis.json \
     --is-genesis-ledger \
-    --root-hash 3NKeMoncuHab5ScarV5ViyF16cJPT4taWNSaTLS64Dp67wuXigPZ \
-    
+    --root-hash 3NKeMoncuHab5ScarV5ViyF16cJPT4taWNSaTLS64Dp67wuXigPZ &
+IDXR_PID=$(pgrep mina-indexer)
```
Comment (trevorbernard) [inline test:87]: I would use `echo $!` over `pgrep`

File: test [hunk 2]
```diff
@@ -36,11 +79,27 @@ idxr server config 2>&1 |
...
+
+# Indexer server starts up on a minimal selection of blocks
+dl_mainnet 50 $TMP/blocks
+server cli \
+    --startup-dir $TMP/blocks \
+    --watch-dir $TMP/blocks \
+    --initial-ledger tests/ledgers/mainnet-genesis.json \
+    --is-genesis-ledger \
+    --root-hash 3NKeMoncuHab5ScarV5ViyF16cJPT4taWNSaTLS64Dp67wuXigPZ &
+IDXR_PID=$(pgrep mina-indexer)
```
Comment (trevorbernard) [inline test:98]: I would use echo $! over pgrep

**Review by trevorbernard — CHANGES_REQUESTED** [review]

**Review by trevorbernard — APPROVED** [review]

**Review by trevorbernard — APPROVED** [review]

**Comment by robinbb** [comment]
@jenr24G This is a very good PR. Thank you!

**Pull request merged**